### PR TITLE
US608063: Generate dotnet schema object builder

### PR DESCRIPTION
**Pull request by kusumaghoshdastidar**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=608063

### CODE REVIEW

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
I know this builds on the build machine, and I can see it building in NetBeans, but it is .NET code so it also must build in Visual Studio.

**NetBeans:**
image

**Visual Studio:**
image

File: dotnet/tests/MicroFocus.FAS.AdapterSdkSchema.Tests/AdapterSdkSchemaObjectBuilderTest.cs [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+            // Datetime
+            documentBuilder.SetDateArchived(DateTime.Now);
+            // Single-valued string
+            documentBuilder.SetAddressBcc("jdoe@gmail.com");
+            // Multi-valued string
+            documentBuilder.SetAddressTo("person1@gmail.com", "person2@there.com");
+
+            // Multi-valued Nested object, Json encoded
+            documentBuilder.SetAccounts(
+                new List<Action<AdapterSdkSchemaObjectBuilder.AccountsObjectBuilder>> {
```
Comment (dermot-hardy) [inline dotnet/tests/MicroFocus.FAS.AdapterSdkSchema.Tests/AdapterSdkSchemaObjectBuilderTest.cs:44]: We should look at adding a set overload that takes a parameter array of actions, to simplify this usage (i.e. to remove the need to construct this list).  We can either merge this and then do a separate PR for it, or just continue with this one.

File: README.md [hunk 2]
```diff
@@ -63,3 +66,141 @@ To use the **.NET** version of the schema add the `MicroFocus.FAS.AdapterSdkSche
...
+    documentBuilder.AddAddressDisplayTo("janeDoe@abc.com");
+```
+
+- Check if all mandatory fields have been set and a valid `AdapterSdkSchemaObject` has been created
+
+```
+    documentBuilder.Validate();
+```
+
+The SchemaObjectBuilder encodes field values as **json** or **flattened**
```
Comment (dermot-hardy) [inline README.md:230]: It's really an internal detail when you're using the object builder, which is sort of the point.  I think it would be more useful to show the usage than to show the output.

**Review by dermot-hardy — APPROVED** [review]

File: dotnet/tests/MicroFocus.FAS.AdapterSdkSchema.Tests/AdapterSdkSchemaObjectBuilderTest.cs (hunk 1 above)
Comment (kusumaghoshdastidar) [reply dotnet/tests/MicroFocus.FAS.AdapterSdkSchema.Tests/AdapterSdkSchemaObjectBuilderTest.cs:44]: Added setter that takes an array

File: README.md (hunk 2 above)
Comment (kusumaghoshdastidar) [reply README.md:230]: Added usage

**Review by dermot-hardy — APPROVED** [review]
